pkg/pillar/pubsub/pubsub.go

```diff
@@ -49,412 +38,137 @@ type keyMap struct {
 	key       *LockedStringMap
 }
 
-// We always publish to our collection.
-// We always write to a file in order to have a checkpoint on process restart.
-// That directory could be persistent in which case it will survive
-// a reboot.
-// The special agent name "" implies always reading from the /var/run/zededa/
-// directory.
-const (
-	publishToSock     = true  // XXX
-	subscribeFromDir  = false // XXX
-	subscribeFromSock = true  // XXX
-
-	// For a subscription, if the agentName is empty we interpret that as
-	// being directory in /var/tmp/zededa
-	fixedName = "zededa"
-	fixedDir  = "/var/tmp/" + fixedName
-	maxsize   = 65535 // Max size for json which can be read or written
-
-	// Copied from types package to avoid cycle in package dependencies
-	// PersistDir - Location to store persistent files.
-	PersistDir = "/persist"
-	// PersistConfigDir is where we keep some configuration across reboots
-	PersistConfigDir = PersistDir + "/config"
-)
-
-func Publish(agentName string, topicType interface{}) (*Publication, error) {
-	return publishImpl(agentName, "", topicType, false)
+// PubSub is a system for publishing and subscribing to messages
+// it manages the creation of Publication and Subscription, which handle the actual
+// implementation of in-memory structures and logic
+// the message passing and persistence are handled by a Driver.
+// Should not be called directly. Instead use the `New()` function.
+type PubSub struct {
+	driver      Driver
+	updaterList *Updaters
```

Comment:
What is the "Updater" ?

Reply:
It is defined in `updaters.go`. It used to be `updaters`, made it exported (capitalized) so that it can be used by different implementations. It is a global list of subscriptions.